docs/keycloak_deployment.md

```diff
@@ -0,0 +1,106 @@
+---
+layout: default
+title: Deploy Keycloak
+nav_order: 3000
+---
+
+# Keycloak Deployment Guide
+
+Keycloak is the service we use for managing users and login sessions. Here we'll cover how to setup a fresh keycloak service from scratch
+
+## Get access to the STAR AWS account
+
+Contact Arend Peter ``@Arend Peter`` on slack to get a AWS user on the STAR AWS account
+
+## Run Instance
+
+1. Go to the EC2 Console
+1. Select "Launch Instance"
+1. Configure as follows
+    * Name: Keycloak Server
+    * OS: Amazon Linux 2 64 Bit (I used ami-0cff7528ff583bf9a but you can use the latest)
+    * Instance Type: t2.micro (Keycloak needs at least .5 GB of memory and this is the cheapest one with 1 GB of memory)
+    * Key Pair: Create Key pair (RSA/pem), and download the pem file (keep the pem file in a safe you'll need it later to access the instance)
+    * Network Settings: Select "Allow HTTPs traffic from the internet"
+    * Configure Storage: Select advanced, 16 GB (default is 8 GB, that might be enough but it just seemed low to me), and set Encrypted to Yes, and KMS Key to aws/ebs
+    * Advanced Details: Set "Stop - Hibernate behaviour" to Enable (this should make it easier to maintain processes in case we need to reboot the server)
+1. Go to view instances, and wait for instance to be in a Running State
+
+## Allocate Elastic IP
+
+By default, EC2 instance will receive a new IP address on every reboot
+
+To keep it the same we need to allocate an elastic IP
+
+1. Within the EC2 console, go to the "Network & Security" section and select "Elastic IPs"
+1. Click "Allocate Elastic IP"
+1. Use the default options, and hit allocate
+1. Under Actions, select "Associate Elastic IP Address"
+1. In the instance section, select the Keycloak instance
+1. Click "Associate"
```

Comment:
maybe note that this ip will be used in several places in the keycloack configuration guide?